### PR TITLE
Fix status during launch, prevent recreating lagObservers

### PR DESCRIPTION
**Pull request by KirillPamPam**

### CODE REVIEW

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum_pos/EthereumPosMultiStream.kt [hunk 1]
```diff
@@ -107,7 +107,7 @@ open class EthereumPosMultiStream(
     }
 
     override fun makeLagObserver(): HeadLagObserver =
-        EthereumPosHeadLagObserver(head, ArrayList(upstreams), headScheduler).apply {
+        EthereumPosHeadLagObserver(head, upstreams, headScheduler).apply {
```
Comment (a10zn8) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum_pos/EthereumPosMultiStream.kt:110]: MutableList it not threadsafe

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/HeadLagObserver.kt [hunk 2]
```diff
@@ -62,11 +62,14 @@ abstract class HeadLagObserver(
             .sample(throttling)
             .flatMap(this::probeFollowers)
             .map { item ->
+                log.debug("Set lag ${item.t1} to upstream ${item.t2.getId()}")
```
Comment (a10zn8) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/HeadLagObserver.kt:65]: Do you want to keep this debug logs?

Comment (KirillPamPam) [reply src/main/kotlin/io/emeraldpay/dshackle/upstream/HeadLagObserver.kt:65]: Yes

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/Multistream.kt [hunk 3]
```diff
@@ -221,6 +221,10 @@ abstract class Multistream(
             }
             when {
                 upstreams.size == 1 -> upstreams[0].setLag(0)
+                    .also {
+                        lagObserver?.stop()
```
Comment (a10zn8) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/Multistream.kt:225]: technically it is better to stop observer first and than set lag to 0

**Review by a10zn8 — APPROVED** [review]

**Pull request merged**